### PR TITLE
feat(config): changelog for the last n commits

### PR DESCRIPTION
**Pull request by FlrnFrmm**
## Description
Adds an additional configuration variable `limit_commits` to the configuration struct.
`limit_commits` can be set to a positive integer number to limit the commits contained in the generated changelog.
Also adjusts the default config file to contain `limit_commits` as a commented-out line.
Extends documentation in README.md to also cover the introduced configuration value.

## Motivation and Context
A possible solution for #102 

## How Has This Been Tested?
Locally with the configuration file `examples/limitedcommits.toml` (added with this PR) and a local git repository.

## Screenshots / Output (if appropriate):
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation (no code change)
- [ ] Refactor (refactoring production code)
- [ ] Other 

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by orhun — CHANGES_REQUESTED** [review]
Looks great, thanks! Only needs a few edits, see my comments 🐻 

Apart from those, I will be doing the following:

- [ ] Mention the `limitedcommits.toml` example in README.md
- [ ] Add test fixtures

Feel free to tackle these too if you'd like!

**Comment by orhun** [comment]
Ah, can you also merge `main` and resolve conflicts?

File: git-cliff/src/lib.rs [hunk 1]
```diff
@@ -217,8 +217,15 @@ pub fn run(mut args: Opt) -> Result<()> {
...
-		repository.commits(commit_range, args.include_path, args.exclude_path)?;
+	let commits = if let Some(commit_limit_value) = config.git.limit_commits {
+		repository
+			.commits(commit_range, args.include_path, args.exclude_path)?
+			.into_iter()
+			.take(commit_limit_value)
+			.collect()
+	} else {
+		repository.commits(commit_range, args.include_path, args.exclude_path)?
+	};
```
[the remark this answers is not in the document]
Comment (FlrnFrmm) [reply git-cliff/src/lib.rs:228]: Oh thanks, that is indeed much nicer :)

**Comment by FlrnFrmm** [comment]
> Ah, can you also merge `main` and resolve conflicts?

Done :+1:

**Comment by FlrnFrmm** [comment]
> Looks great, thanks! Only needs a few edits, see my comments bear
> 
> Apart from those, I will be doing the following:
> 
> * [ ]  Mention the `limitedcommits.toml` example in README.md
> * [ ]  Add test fixtures
> 
> Feel free to tackle these too if you'd like!

Added it to the README.md.

Was not sure where to find the test fixtures and what to do there :disappointed:

**Pull request merged**